### PR TITLE
OP-1168 Add LogViewer

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1168.

### CODE REVIEW

File: src/main/java/org/isf/utils/log/LogUtil.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			return absolutePath;
+		} else {
+			return "No appender found with the name 'RollingFile' in the root logger.";
+		}
+	}
+
+	/**
+	 * Opens the application log file ({@code openhospital.log}) folder with the System default file explorer.
+	 *
+	 * @param path the path
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/utils/log/LogUtil.java:56]: This line should be removed; there are no parameters to this method

File: src/main/java/org/isf/utils/log/LogUtil.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	/**
+	 * Gets the application log file ({@code openhospital.log}) absolute path defined in the {@code log4j.properties}
+	 * 
+	 * @return
+	 */
+	public static String getLogFileAbsolutePath() {
+		Logger rootLogger = Logger.getRootLogger();
+		Appender appender = rootLogger.getAppender("RollingFile");
+
+		if (appender != null && appender instanceof DailyRollingFileAppender) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/utils/log/LogUtil.java:43]:
```suggestion
		if (appender instanceof DailyRollingFileAppender) {
```
the `instanceof` handles `null` arguments

File: src/main/java/org/isf/utils/log/LogUtil.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.log4j.Appender;
+import org.apache.log4j.DailyRollingFileAppender;
+import org.apache.log4j.Logger;
+
+public class LogUtil {
+
+	/**
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/utils/log/LogUtil.java:34]: This method only has static methods (it is a util class after all) so there should be a `private` constructor to stop someone trying to instantiate this class.

File: src/main/java/org/isf/utils/log/LogUtil.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/utils/log/LogUtil.java:56]: Thanks

File: src/main/java/org/isf/utils/log/LogUtil.java (hunk 2 above)
Comment (mwithi) [reply src/main/java/org/isf/utils/log/LogUtil.java:43]: Thanks!

File: src/main/java/org/isf/utils/log/LogUtil.java (hunk 3 above)
Comment (mwithi) [reply src/main/java/org/isf/utils/log/LogUtil.java:34]: nice tip! thanks!

**Review by dbmalkovsky — DISMISSED** [review]
LGTM; one little "improvement" 😏

File: src/main/java/org/isf/utils/log/LogUtil.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import java.io.IOException;
+
+import org.apache.log4j.Appender;
+import org.apache.log4j.DailyRollingFileAppender;
+import org.apache.log4j.Logger;
+
+public class LogUtil {
+
+	private LogUtil() {
+		super();
```
[the remark this answers is not in the document]
Comment (mwithi) [reply src/main/java/org/isf/utils/log/LogUtil.java:35]: well, it always extends the Object class... 😛 I liked more than an empty constructor, but removed anyway.
Thanks!

**Review by mizzioisf — APPROVED** [review]
LGTM, I think we can merge. I will test "in production" before 1.14.0 release.

**Review by dbmalkovsky — APPROVED** [review]
LGTM

**Pull request merged**